tests/stdlib/thttpclient_standalone.nim

```diff
@@ -0,0 +1,32 @@
+discard """
+  cmd: "nim c --threads:on $file"
+  output: "OK"
+"""
+
+import asynchttpserver, httpclient, asyncdispatch
+
+block: # issue #16436
+  proc startServer() {.async.} =
+    proc cb(req: Request) {.async.} =
+      let headers = { "Content-length": "15"} # Provide invalid content-length
+      await req.respond(Http200, "Hello World", headers.newHttpHeaders())
+
+    var server = newAsyncHttpServer()
+    await server.serve(Port(5555), cb)
+
+  proc runClient() {.async.} =
+    let c = newAsyncHttpClient(headers = {"Connection": "close"}.newHttpHeaders)
+    let r = await c.getContent("http://127.0.0.1:5555")
+    echo r
+
+  asyncCheck startServer()
+
+  var gotException = false
+  try:
+    waitFor runClient()
+  except ProtocolError:
+    gotException = true
+
+  doAssert(gotException)
+
+echo "OK"
```

Comment:
remove this line and `output: "OK"` and use `cmd: "nim c -r --threads:on $file"` to be on safe side that this file runs (even though it will still run with `cmd: "nim c --threads:on $file"`)

refs: https://github.com/nim-lang/Nim/pull/16678#discussion_r555446076